### PR TITLE
Add test for two empty strings in binary comparator

### PR DESCRIPTION
**Pull request by gramian**
## What does this PR do?
This change adds a check for two empty strings to be compared to the `equalsString` method of the `BinaryComparator` class. As far as I understand it, strings are compared as byte arrays which can cause the observed error when comparing the last byte. However for a length of zero, as an empty string has, this causes an error.

Now I cannot say why this error does not happen with the current repo head, but I think it should or still might, thus I added this check for both argument strings being empty and preventing this error.

## Related issues
https://github.com/ArcadeData/arcadedb/issues/1297

## Additional Notes
This specifically does not add the check to the `equalsBytes` as I do not know if a zero length array can actually be created.

## Checklist
- [X] I have run the build using `mvn clean package` command
- [ ] My unit tests cover both failure and success scenarios

### CODE REVIEW

File: engine/src/main/java/com/arcadedb/serializer/BinaryComparator.java [hunk 1]
```diff
@@ -396,6 +396,9 @@ public static boolean equalsString(final String buffer1, final String buffer2) {
     if (buffer1 == null || buffer2 == null)
       return false;
 
+    if (buffer1.equals("") && buffer2.equals(""))
```
Comment (lvca) [inline engine/src/main/java/com/arcadedb/serializer/BinaryComparator.java]: I could be more efficient doing `if( buffer1.isEmpty() && buffer2.isEmpty() )`

Comment (gramian) [reply engine/src/main/java/com/arcadedb/serializer/BinaryComparator.java]: Done.

**Review by lvca — APPROVED** [review]
